src/Halcyon/Datasource/FileDatasource.php

```diff
@@ -337,6 +342,10 @@ public function getAvailablePaths(): array
             ? new RecursiveIteratorIterator(new RecursiveDirectoryIterator($this->basePath))
             : [];
 
+        if (!is_array($it)) {
+            $it->setMaxDepth($this->maxDepth + 1);
+        }
```

Comment:
What does this do?

Reply:
Sets the iterator maxDepth on the iterator used inside `getAvailablePaths()`. This is to ensure that the reported paths match the same reported paths in `select()` as both methods implement their own iterators